### ISSUE TITLE
ChatOpenAI being loaded despite using settings with ChatAnthropic

### ISSUE THREAD

**Issue by lukestanley** [ju-bezdek/langchain-decorators#2]
Maybe I'm using it wrong but there is an error after I try setting the LLM in both ways shown in the README.

```
from langchain.chat_models import ChatAnthropic

from langchain_decorators import GlobalSettings

GlobalSettings.define_settings(
    default_llm=ChatAnthropic(temperature=0.0),
    default_streaming_llm=ChatAnthropic(temperature=0.0,streaming=True),
)

PromptTypes.AGENT_REASONING.llm = ChatAnthropic()
```

```Traceback (most recent call last):
  File "/workspaces/langchain-decorators/src/test.py", line 6, in <module>
    from langchain_decorators import GlobalSettings
  File "/workspaces/langchain-decorators/src/langchain_decorators/__init__.py", line 1, in <module>
    from .common import LogColors, GlobalSettings, print_log, PromptTypes, PromptTypeSettings
  File "/workspaces/langchain-decorators/src/langchain_decorators/common.py", line 12, in <module>
    class GlobalSettings(BaseModel):
  File "/workspaces/langchain-decorators/src/langchain_decorators/common.py", line 27, in GlobalSettings
    default_llm=ChatOpenAI(temperature=0.0),
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for ChatOpenAI
__root__
  Did not find openai_api_key, please add an environment variable `OPENAI_API_KEY` which contains it, or pass  `openai_api_key` as a named parameter. (type=value_error)```

**Comment by ju-bezdek** [comment]
It looks correct to me... will look into it

**Comment by ju-bezdek** [comment]
Oh... obviously ... I put the ChatOpenAI as a default into the function declaration, so it initializes during import...
I didn't realize that langchain validates this before invoking the LLM...

gonna fix it, it he meantime you can just define any OPENAI_API_KEY in your ENVs ... should work since you are overriding the default settings (your code is invoked later)

**Comment by lukestanley** [comment]
I have a fix. Sorting PR now.

**Comment by lukestanley** [comment]
It would be best to sort out the code formatting before so the diff is more obvious. Sorry I didn't split it into two commits!
https://github.com/ju-bezdek/langchain-decorators/pull/3/files
@ju-bezdek

**Comment by lukestanley** [comment]
I used Anthropic's big context window to look at your files to quickly suggest the fix:
https://gist.github.com/lukestanley/ddfd48489ecb5bc97e4d2bca24ca229f ;)

**Comment by ju-bezdek** [comment]
That's super cool 🤣🚀
gonna review your PR now

**Comment by ju-bezdek** [comment]
fixed in v0.0.3
